Algorithm/QCAlgorithm.History.cs

```diff
@@ -608,7 +608,10 @@ private IEnumerable<HistoryRequest> CreateDateRangeHistoryRequests(IEnumerable<S
 
                     // apply overrides
                     var res = GetResolution(x, resolution);
-                    if (fillForward.HasValue) request.FillForwardResolution = fillForward.Value ? res : null;
+                    if (fillForward.HasValue)
+                    {
+                        request.FillForwardResolution = (fillForward.Value && resolution != Resolution.Tick) ? res : null;
```

Comment:
Wondering if this should be handled in the algorithm layer or should be handled via the `IHistoryProvider` implementation, or maybe even better, the `HistoryRequest` constructor could manage this -- and then it's in a single spot.

Reply:
Good idea! The properties are being set outside of the constructor, so I'll add an internal check to the `HistoryRequest.FillForwardResolution { get; }`. Why not the `{ set }`? Because it depends on the value of `HistoryRequest.Resolution`, but that value can change after the `FillForwardResolution` was set.